includes/class-wc-payments-order-service.php

```diff
@@ -1072,4 +1268,71 @@ private function update_order_status( $order, $order_status, $intent_id = '' ) {
 			Logger::log( 'Error when updating status for order ' . $order->get_id() . ': ' . $e->getMessage() );
 		}
 	}
+
+	/**
+	 * Takes an intent object or array and returns our needed data as an array.
+	 * This is needed due to intents can either be objects or arrays.
+	 *
+	 * @param object|array $intent The intent to pull the data from.
+	 *
+	 * @return array The data we need to continue processing.
+	 */
+	private function get_intent_data( $intent ): array {
+		$intent_data = [];
+		if ( is_array( $intent ) ) {
+			$intent_data = [
+				'intent_id'     => $intent['id'],
+				'intent_status' => $intent['status'],
+				'charge_id'     => $intent['charge_id'] ?? '',
+				'fraud_outcome' => $intent['fraud_outcome'] ?? '',
+			];
+		} elseif ( is_object( $intent ) ) {
+			$charge      = $intent->get_charge();
+			$intent_data = [
+				'intent_id'     => $intent->get_id(),
+				'intent_status' => $intent->get_status(),
+				'charge_id'     => $charge ? $charge->get_id() : null,
+				'fraud_outcome' => $intent->get_metadata()['fraud_outcome'] ?? '',
+			];
+		}
+		return $intent_data;
+	}
```

Comment:
This method was created due to we have intents that can either be objects or arrays, and they are used interchangeably. 

Brought this up to the team: p1677173243169499-slack-C01B8KNUYSW